taskana-adapter-camunda-outbox-rest/src/main/java/pro/taskana/adapter/camunda/outbox/rest/service/CamundaTaskEventsService.java

```diff
@@ -0,0 +1,252 @@
+    private static final String SQL_GET_COMPLETE_AND_DELETE_EVENTS = "SELECT * FROM taskana_tables.event_store WHERE type = ? OR type = ?";
+    private static final String SQL_WITHOUT_PLACEHOLDERS_DELETE_EVENTS = "DELETE FROM taskana_tables.event_store WHERE id in (%s)";
+
+    public List<CamundaTaskEventResource> getEvents(List<String> requestedEventTypes) {
+
+        List<CamundaTaskEventResource> camundaTaskEventResources = new ArrayList<>();
+
+        if (requestedEventTypes.contains("create")) {
+
+            camundaTaskEventResources = getCreateEvents();
+
+        } else if (requestedEventTypes.contains("delete") && requestedEventTypes.contains("complete")) {
+
+            camundaTaskEventResources = getCompleteAndDeleteEvents();
+        }
+
+        return camundaTaskEventResources;
+
+    }
+
+    public void deleteEvents(String idsAsJsonArray) {
+
+        List<Integer> idsAsIntegers = getIdsAsIntegers(idsAsJsonArray);
+
+        String DeleteEventsSqlWithPlaceholders = String.format(SQL_WITHOUT_PLACEHOLDERS_DELETE_EVENTS, preparePlaceHolders(idsAsIntegers.size()));
+
+        try (Connection connection = getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(DeleteEventsSqlWithPlaceholders)
+        ) {
+
+            setPreparedStatementValues(preparedStatement, idsAsIntegers);
+            preparedStatement.execute();
+
+        } catch (SQLException e) {
+            LOGGER.warn("Caught {} while trying to delete events from the outbox table", e);
+        }catch (Exception e){
+            LOGGER.warn("Caught {} while trying to delete events from the outbox table", e);
+        }
+
+    }
+
+    private List<CamundaTaskEventResource> getCreateEvents() {
+
+        List<CamundaTaskEventResource> camundaTaskEventResources = new ArrayList<>();
+
+        try (
+                Connection connection = getConnection();
+                ResultSet createEventsResultSet = prepareAndExecuteCreateEventsQuery(connection)
+        ) {
+
+            camundaTaskEventResources = getCamundaTaskEventResources(createEventsResultSet);
+
+        } catch (SQLException e) {
+            LOGGER.warn("Caught {} while trying to retrieve create events from the outbox", e);
+        } catch (NullPointerException e) {
+            LOGGER.warn("Caught {} while trying to retrieve create events from the outbox", e);
+        }
+        return camundaTaskEventResources;
+    }
+
+
+    private String preparePlaceHolders(int length) {
+        return String.join(",", Collections.nCopies(length, "?"));
+
+    }
+
+    private void setPreparedStatementValues(PreparedStatement preparedStatement, List<Integer> ids) throws SQLException {
+        for (int i = 0; i < ids.size(); i++) {
+            preparedStatement.setObject(i + 1, ids.get(i));
+        }
+    }
+
+
+    private List<CamundaTaskEventResource> getCamundaTaskEventResources(ResultSet createEventsResultSet) throws SQLException, NullPointerException {
+
```

Comment:
NullpointerException is a runtime exception - should not be declared as thrown

Reply:
changed